haskell/src/Stablecoin/Client/Cleveland/Caps.hs

```diff
@@ -0,0 +1,229 @@
+mint = actionToCaps siMint
+
+burn
+  :: MonadStablecoin caps base m
+  => "sender" :! AddressOrAlias -> "contract" :! AddressOrAlias
+  -> NonEmpty Natural -> m ()
+burn = actionToCaps siBurn
+
+transferOwnership
+  :: MonadStablecoin caps base m
+  => "sender" :! AddressOrAlias -> "contract" :! AddressOrAlias
+  -> AddressOrAlias -> m ()
+transferOwnership = actionToCaps siTransferOwnership
+
+acceptOwnership
+  :: MonadStablecoin caps base m
+  => "sender" :! AddressOrAlias -> "contract" :! AddressOrAlias
+  -> m ()
+acceptOwnership = actionToCaps siAcceptOwnership
+
+changeMasterMinter
+  :: MonadStablecoin caps base m
+  => "sender" :! AddressOrAlias -> "contract" :! AddressOrAlias
+  -> AddressOrAlias -> m ()
+changeMasterMinter = actionToCaps siChangeMasterMinter
+
+changePauser
+  :: MonadStablecoin caps base m
+  => "sender" :! AddressOrAlias -> "contract" :! AddressOrAlias
+  -> AddressOrAlias -> m ()
+changePauser = actionToCaps siChangePauser
+
+setTransferlist
+  :: MonadStablecoin caps base m
+  => "sender" :! AddressOrAlias -> "contract" :! AddressOrAlias
+  -> Maybe AddressOrAlias -> m ()
+setTransferlist = actionToCaps siSetTransferlist
+
+getStablecoinBalance :: MonadStablecoin caps base m => "contract" :! AddressOrAlias -> m Mutez
+getStablecoinBalance = actionToCaps siGetStablecoinBalance
+
+getPaused :: MonadStablecoin caps base m => "contract" :! AddressOrAlias -> m Bool
+getPaused = actionToCaps siGetPaused
+
+getContractOwner :: MonadStablecoin caps base m => "contract" :! AddressOrAlias -> m AddressAndAlias
+getContractOwner = actionToCaps siGetContractOwner
+
+getPendingContractOwner :: MonadStablecoin caps base m => "contract" :! AddressOrAlias -> m (Maybe AddressAndAlias)
+getPendingContractOwner = actionToCaps siGetPendingContractOwner
+
+getMasterMinter :: MonadStablecoin caps base m => "contract" :! AddressOrAlias -> m AddressAndAlias
+getMasterMinter = actionToCaps siGetMasterMinter
+
+getPauser :: MonadStablecoin caps base m => "contract" :! AddressOrAlias -> m AddressAndAlias
+getPauser = actionToCaps siGetPauser
+
+getTransferlist :: MonadStablecoin caps base m => "contract" :! AddressOrAlias -> m (Maybe AddressAndAlias)
+getTransferlist = actionToCaps siGetTransferlist
+
+getTotalSupply :: MonadStablecoin caps base m => "contract" :! AddressOrAlias -> m Natural
+getTotalSupply = actionToCaps siGetTotalSupply
+
+assertEq :: MonadStablecoin caps base m => (Eq a, Show a) => a -> a -> m ()
+assertEq = actionToCaps siAssertEq
+
+-- TODO: this already exists in Cleveland, but it's not exported.
+-- We should export it.
+revealKeyUnlessRevealed :: MonadStablecoin caps base m => Alias -> m ()
+revealKeyUnlessRevealed = actionToCaps siRevealKeyUnlessRevealed
+
+----------------------------------------------------------------------------
+-- Internal helpers
+----------------------------------------------------------------------------
+
+-- TODO: this already exists in Cleveland, but it's not exported.
```

Comment:
Regarding `caps`, I have just created an issue I wanted to create long ago: https://gitlab.com/morley-framework/morley/-/issues/309. No need to change anything here, just sharing some thoughts.